File: 36.py
from typing import BinaryIO


def palindrome(val) -> bool:
    if type(val) != str:
        val = str(val)
    fwd_cnt = 0
    bck_cnt = len(val) - 1
    while fwd_cnt < bck_cnt:
        if val[fwd_cnt] != val[bck_cnt]:
            return False
        fwd_cnt += 1
        bck_cnt -= 1
    return True

def to_bin(base10) -> str:
    return bin(base10)[2:]


total = 0
for x in range(1000000):
    if palindrome(x):
        if palindrome(to_bin(x)):
            total += x

print(total)

File: 36.cpp
#include <iostream>
#include <string>

bool palindrome(std::string val) {
  std::string::iterator fwd_iter = val.begin();
  std::string::reverse_iterator bck_iter = val.rbegin();
  std::string::size_type half = val.size() / 2;
  std::string::size_type counter = 0;

  while (counter < half) {
    if (*fwd_iter != *bck_iter) {
      return false;
    }
    ++fwd_iter;
    ++bck_iter;
    ++counter;
  }

  return true;
}

std::string toBin(int num) {
  std::string buffer;
  while (num != 0) {
    buffer = (num % 2 == 0 ? "0" : "1") + buffer;
    num /= 2;
  }
  return buffer;
}

int main() {
  int total = 0;
  for (int i = 1; i < 1000000; ++i) {
    std::string stri = std::to_string(i);
    if (palindrome(stri)) {
      if (palindrome(toBin(i))) {
        total += i;
      };
    }
  }
  std::cout << total << std::endl;
  return 0;
}